packages/block-editor/src/components/block-draggable/draggable-chip.js

```diff
@@ -13,7 +13,10 @@ import BlockIcon from '../block-icon';
 export default function BlockDraggableChip( { count, icon, isPattern } ) {
 	const patternLabel = isPattern && __( 'Pattern' );
 	return (
-		<div className="block-editor-block-draggable-chip-wrapper">
+		<div
+			className="block-editor-block-draggable-chip-wrapper"
+			style={ { opacity: 'var(--wp--block-draggable-valid-opacity)' } }
```

Comment:
Why is this applied via an inline style and CSS variable instead of by classname and direct value in `style.scss`? Or to put it slightly differently, does the JS version of the component need to have an opinion about how it's styled?

Reply:
Good question! I don't think it has to be applied as an inline style; I initially added it that way for the sake of development speed. It _does_ have to be a CSS custom property, because its value needs to change when the body class gets added, and given that the chip we see on the page is [a stored copy of itself](https://github.com/WordPress/gutenberg/pull/56843#discussion_r1424849732) changing classes on the chip itself won't work. Essentially the custom property is acting like a ref to the element, so we can manipulate its value from outside.

Reply:
> It does have to be a CSS custom property, because its value needs to change when the body class gets added

Oh, I was imagining something like the following:

```
.block-editor-block-draggable-chip {
	opacity: 1;
}

.block-draggable-invalid-drag-token {
	.block-editor-block-draggable-chip {
		opacity: 0.6;
	}
}
```

Or does that not work? (Totally fine to go with a CSS variable by the way, just trying to understand what the limitations are 🙂)

Reply:
Oh right, yeah that should work too! Would that be preferable to a custom property? I don't think specificity matters either way.

Reply:
I don't feel strongly either way, but I'd lean slightly more toward the nested selectors so that it's easy to swap out for (or combine) other styles like background color, border, or other subtle styling changes that folks might want to explore in the future.

Reply:
Oh good point, yeah that would make it easier to change styles altogether if we wanted to.